### PR TITLE
themis/backfill from block [TAC-1160]

### PR DESCRIPTION
**Pull request by 0xThemis**
- **feat(key-gen): add explicit block to start backfill**
- **test: added explicit backfill block test**

---

> [!NOTE]
> **Medium Risk**
> Changes event replay start position on startup, which can reprocess historical key-gen events if misconfigured; persisted cursor still wins when it is at or after the explicit block.
> 
> **Overview**
> Adds optional **`explicit_backfill_block`** config so key-gen can replay registry events from a chosen block when the persisted chain cursor is still behind that point.
> 
> On startup, **`select_backfill_cursor`** compares the loaded cursor with the configured block: if the cursor is earlier, it rewinds to the end of the preceding block (`block - 1`, `u64::MAX`) so inclusive backfill includes log index 0; if the cursor is already at or past that block, the explicit setting is ignored.
> 
> The option is wired through service config, **`key_event_watcher`**, and test helpers; unit tests cover the three cursor-selection cases, and an integration test verifies keygen when **`init_keygen`** runs before nodes start (interval mining + explicit backfill).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 2aca5ed58f01fbf41e9d23226131ebce00774c68. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: oprf-key-gen/src/services/key_event_watcher/tests.rs [hunk 1]
```diff
@@ -389,3 +398,30 @@ async fn test_abort() -> eyre::Result<()> {
     );
     Ok(())
 }
+
+#[test]
+fn explicit_backfill_starts_before_configured_block() {
```
Comment (dkales) [inline oprf-key-gen/src/services/key_event_watcher/tests.rs:403]: maybe also add a test to check against the return type of load against an empty DB, since that is the case we want to use most I guess.

**Review by dkales — APPROVED** [review]
